pkg/server/commands/listusers/list_users.go

```diff
@@ -13,64 +13,78 @@ type listUsersRequest interface {
 	GetRelation() string
 	GetUserFilters() []*openfgav1.ListUsersFilter
 	GetContextualTuples() *openfgav1.ContextualTupleKeys
+	GetDepth() uint32
 }
 
-type internalListUsersRequest struct {
+type ListUsersRequest struct {
 	*openfgav1.ListUsersRequest
 
 	// visitedUsersetsMap keeps track of the "path" we've made so far.
 	// It prevents stack overflows by preventing visiting the same userset twice.
 	visitedUsersetsMap map[string]struct{}
+
+	// Depth is the current depths of the traversal expressed as a positive, decrementing integer.
+	// When expansion of list users recursively traverses one level, we decrement by one. If this
+	// counter is 0, we throw ErrResolutionDepthExceeded. This protects against a potentially deep
+	// or endless cycle of recursion.
+	Depth uint32
```

Comment:
This is personal preference so i'm not going to block, but can we do it how ListObjects does it? Start with depth:0 and then increase it on each level, when it hits the maximum allowed, you exit. It is more intuitive to me.

Reply:
While I agree with the algorithm described in the comments, if ListObjects does it the other way around, I'd advocate for consistency as @miparnisari suggests.

Reply:
I personally prefer the incrementing approach so I'm going to change to that. Hopefully we'll make Check consistent in the future.

Reply:
Yeah we have a task for it: https://github.com/openfga/openfga/issues/994